c++/src/ColumnReader.cc

```diff
@@ -1006,9 +1009,12 @@ namespace orc {
     if (stream == nullptr)
       throw ParseError("LENGTH stream not found in List column");
     rle = createRleDecoder(std::move(stream), false, vers, memoryPool);
-    const Type& childType = *type.getSubtype(0);
-    if (selectedColumns[static_cast<uint64_t>(childType.getColumnId())]) {
-      child = buildReader(childType, stripe);
+    ArrayReadIntent intent = stripe.getReadIntent(columnId);
+    if (intent == ArrayReadIntent_ALL) {
+      const Type &childType = *type.getSubtype(0);
+      if (selectedColumns[static_cast<uint64_t>(childType.getColumnId())]) {
+        child = buildReader(childType, stripe);
+      }
```

Comment:
Do we have a test coverage for `else` case? Could you point me the test case for `ArrayReadIntent_OFFSETS`?

Reply:
It is tested at [TestColumnReaderEncoded.testListOffsetsOnly](https://github.com/apache/orc/pull/990/files#diff-e0eca596eb78b0688bdb1d04e6a23d5b7c101ff686e99050dbe15147d275ec47R1474), which is modified from TestColumnReaderEncoded.testList.

The assertion is
```
ASSERT_EQ(0, longs->numElements);
```

Reply:
I replace this with 4 new tests at `c++/test/TestReader.cc`.
However, they stop at column selection verification and do not read a row batch.